qute.jdt/com.redhat.qute.jdt/src/main/java/com/redhat/qute/jdt/internal/extensions/renarde/RenardeResolvedJavaTypeFactory.java

```diff
@@ -0,0 +1,159 @@
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.eclipse.jdt.core.IAnnotation;
+import org.eclipse.jdt.core.IField;
+import org.eclipse.jdt.core.ILocalVariable;
+import org.eclipse.jdt.core.IMethod;
+import org.eclipse.jdt.core.IType;
+import org.eclipse.jdt.core.JavaModelException;
+
+import com.redhat.qute.commons.InvalidMethodReason;
+import com.redhat.qute.commons.JavaMethodInfo;
+import com.redhat.qute.commons.ResolvedJavaTypeInfo;
+import com.redhat.qute.commons.datamodel.resolvers.ValueResolverKind;
+import com.redhat.qute.commons.jaxrs.JaxRsMethodKind;
+import com.redhat.qute.commons.jaxrs.JaxRsParamKind;
+import com.redhat.qute.commons.jaxrs.RestParam;
+import com.redhat.qute.jdt.internal.resolver.ITypeResolver;
+import com.redhat.qute.jdt.internal.template.resolvedtype.AbstractResolvedJavaTypeFactory;
+import com.redhat.qute.jdt.utils.AnnotationUtils;
+
+/**
+ * Custom Factory to create an {@link ResolvedJavaTypeInfo} instance for Renarde
+ * controller.
+ * 
+ * @author Angelo ZERR
+ *
+ */
+public class RenardeResolvedJavaTypeFactory extends AbstractResolvedJavaTypeFactory {
+
+	private static final String JAVAX_WS_RS_POST_ANNOTATION = "javax.ws.rs.POST";
+	private static final String JAKARTA_WS_RS_POST_ANNOTATION = "jakarta.ws.rs.POST";
+
+	private static final String REST_FORM_ANNOTATION = "org.jboss.resteasy.reactive.RestForm";
+
+	private static final String JAVAX_WS_RS_FORM_PARAM_ANNOTATION = "javax.ws.rs.FormParam";
+
+	private static final String JAKARTA_WS_RS_FORM_PARAM_ANNOTATION = "jakarta.ws.rs.FormParam";
+
+	private static final String REST_PATH_ANNOTATION = "org.jboss.resteasy.reactive.RestPath";
+
+	private static final String JAVAX_WS_RS_PATH_PARAM_ANNOTATION = "javax.ws.rs.PathParam";
+
+	private static final String JAKARTA_WS_RS_PATH_PARAM_ANNOTATION = "jakarta.ws.rs.PathParam";
+
+	private static final String REST_QUERY_ANNOTATION = "org.jboss.resteasy.reactive.RestQuery";
+
+	private static final String JAVAX_WS_RS_QUERY_PARAM_ANNOTATION = "javax.ws.rs.QueryParam";
+
+	private static final String JAKARTA_WS_RS_QUERY_PARAM_ANNOTATION = "jakarta.ws.rs.QueryParam";
+
+	@Override
+	public boolean isAdaptedFor(ValueResolverKind kind) {
+		return kind == ValueResolverKind.Renarde;
+	}
+
+	@Override
+	protected boolean isValidField(IField field, IType type) throws JavaModelException {
+		return false;
+	}
+
+	@Override
+	protected boolean isValidRecordField(IField field, IType type) {
+		return false;
+	}
+
+	@Override
+	protected InvalidMethodReason getValidMethodForQute(IMethod method, String typeName) {
+		return null;
+	}
+
+	@Override
+	protected JavaMethodInfo createMethod(IMethod method, ITypeResolver typeResolver) {
+		JavaMethodInfo info = super.createMethod(method, typeResolver);
+		collectJaxrsInfo(method, info);
+		return info;
+	}
+
+	private static void collectJaxrsInfo(IMethod method, JavaMethodInfo info) {
+		JaxRsMethodKind methodKind = JaxRsMethodKind.GET;

```

Comment:
The main goal is to mark the method as renarde method controllerto manage custom validation on Qute LS. To avoid having a flag JavaMethodInfo#isRenarde(), I use JaxRsMethodKind because perhaps in the future we could have some specific rule for @POST, @GET (ex: you cannot use the @POST method outside an HTML form)

I wanted just to distinguish POST and GET, but other kind are not supported because we don't need it for the moment.

Reply:
Okay. That makes sense to me. In the future it would be interesting to support `@DELETE` and `@PUT` if we want to provide integration with HTMX like in the [renotes sample project](https://github.com/ia3andy/renotes), but for now I am okay with just distinguishing `@POST` from the others.

Reply:
fixed

Reply:
Sorry, I should have been more clear. I meant in the `enum` class

Reply:
ok, fixed